blockchain/state_transition.go

```diff
@@ -363,6 +363,19 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	)
 	ret, st.gas, vmerr = msg.Execute(st.evm, st.state, st.evm.Context.BlockNumber.Uint64(), st.gas, st.value)
 
+	// These tx types does not enter the EVM in the msg.Execute() method. For the purpose of debug traces,
+	// those tx types are considered as a harmless zero-value transfer to sender itself. This aligns with
+	// how the eth_getTransaction fills the 'to' field for those tx types. See also api_ethereum.go:resolveToField
+	if st.evm.Config.Debug {
+		txType := msg.Type()
+		if txType.IsAccountUpdate() || txType.IsCancelTransaction() || txType.IsChainDataAnchoring() {
+			st.evm.Config.Tracer.CaptureStart(st.evm, msg.ValidatedSender(), msg.ValidatedSender(), false, msg.Data(), st.initialGas, msg.Value())
+			defer func() {
+				st.evm.Config.Tracer.CaptureEnd(ret, st.gas, vmerr)
```

Comment:
Don't we need to pass gasUsed?
```suggestion
				st.evm.Config.Tracer.CaptureEnd(ret, st.gasUsed(), vmerr)
```

Reply:
The second argument is called `restGas`, a.k.a. remaining gas, leftover gas. If you see the call_tracer.go, the resulting gasUsed is calculated as a difference between the gasLimit and restGas.
Also refer to the tx [0xac43859eb4064916e8be8e74645d6019cc48cb6791f68ea21d42ead6bba569b5](https://baobab.klaytnfinder.io/tx/0xac43859eb4064916e8be8e74645d6019cc48cb6791f68ea21d42ead6bba569b5) where its gasUsed is correctly returned.

Reply:
Ok, I'd understand it's ok. But the second argument is currently called `gasUsed`  I think: https://github.com/kaiachain/kaia/blob/dev/blockchain/vm/logger.go#L113

Reply:
Oh, you are right. It must be `st.gasUsed()`. I confused with CaptureTxEnd(restGas).
The output was okay nonetheless because the top call frame's gasUsed is filled in CaptureTxEnd. So the CaptureEnd's gasUsed argument is essentially ignored in CallTracer.
But for other tracers that might use the field, I'll fix it.